### PR TITLE
A few more fixes related to filenames and content types

### PR DESCRIPTION
**Pull request by matiaskorhonen**
- Fixes the redirect Location for certain files
- Prevents an exception in the special case that no MIME type is found
- Uses the "recombined" name everywhere

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -64,22 +65,29 @@ class S3itchApp < Sinatra::Base
...
         content_type: content_type,
         metadata: { "Cache-Control" => 'public, max-age=315360000'}
       })
-      puts "Uploaded file #{params[:name]} to S3"
+      puts "Uploaded file #{file.key} to S3"
       if ENV['NO_CNAME']
         redirect "#{file.public_url}", 201
       else
-        redirect "#{ENV['S3_BUCKET']}/#{file.key}", 201
+        redirect "http://#{ENV['S3_BUCKET']}/#{file.key}", 201
```
Comment (matiaskorhonen) [inline app.rb:87]: Not having the URI scheme breaks the redirect for some clients (namely my own [s3itch_client](https://github.com/k33l0r/s3itch_client)).

**Comment by matiaskorhonen** [comment]
How about getting this merged?

**Comment by matiaskorhonen** [comment]
If it helps, I'm using this branch myself and everything seems to work just swimmingly.

**Comment by roidrage** [comment]
Thanks!

**Pull request merged**